From 95653cbfda50d8f4a5d406990a04ed3584f5eb5e Mon Sep 17 00:00:00 2001
From: Vincent Abriou <vincent.abriou@st.com>
Date: Fri, 7 Oct 2016 17:03:59 +0200
Subject: [PATCH 30/61] gl-renderer: conditionally call query_buffer while
 gl_renderer_attach

While gl_renderer_attach, query_buffer should be call only if the
query_buffer function exists.

Change-Id: I5b5995fec67351cc8e4746f932d7ace838fa73be
Signed-off-by: Vincent Abriou <vincent.abriou@st.com>
---
 libweston/gl-renderer.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libweston/gl-renderer.c b/libweston/gl-renderer.c
index 9747de5..69b0e4a 100644
--- a/libweston/gl-renderer.c
+++ b/libweston/gl-renderer.c
@@ -2004,7 +2004,8 @@ gl_renderer_attach(struct weston_surface *es, struct weston_buffer *buffer)
 
 	if (shm_buffer)
 		gl_renderer_attach_shm(es, buffer, shm_buffer);
-	else if (gr->query_buffer(gr->egl_display, (void *) buffer->resource,
+	else if (gr->query_buffer &&
+		 gr->query_buffer(gr->egl_display, (void *)buffer->resource,
 				  EGL_TEXTURE_FORMAT, &format))
 		gl_renderer_attach_egl(es, buffer, format);
 	else if ((dmabuf = linux_dmabuf_buffer_get(buffer->resource)))
-- 
2.7.4

